subprojects/docs/src/docs/userguide/managingTransitiveDependencies.adoc

```diff
@@ -30,6 +30,13 @@ Dependency constraints allow you to define the version or the version range of b
 
 In the example, all versions are omitted from the dependency declaration. Instead, the versions are defined in the constraints block. The version definition for `commons-codec:1.11` is only taken into account if `commons-codec` is brought in as transitive dependency, since `commons-codec` is not defined as dependency in the project. Otherwise, the constraint has no effect.
 
+Dependency constraints themselves can also be added transitively. If a modules's metadata is defined in a `.pom` file that contains dependency entries with `<optional>true</optional>`, Gradle will create a dependency constraint for each of these so-called _optional dependencies_. This leads to a similar resolution behavior as provided by Maven: if the corresponding module is brought in by another, non-optional, dependency declaration, then the constraint will apply when choosing the version for that module (e.g., if the optional dependency defines a higher version, that one is chosen).
```

Comment:
Can you make the title of this section use-case driven while you are at it?

Reply:
Ha. I thought you already did that with another PR... 
I changed it.